### DISCUSSION TITLE
Insert after jump expected behaviour

### DISCUSSION THREAD

**Discussion by willparsons** [helix-editor/helix#3500] [Q&A] [answered]
After using an action like `w,f,t` the cursor moves to the appropriate place but pressing `i` moves the cursor back to where it was before the original move action.

Is this expected behaviour?

**Comment by the-mikedavis** [comment, marked as answer]
Yep that's the expected behavior: <kbd>i</kbd> inserts at the beginning of the selection. <kbd>w</kbd>/<kbd>f</kbd>/<kbd>t</kbd> and others create a selection that starts at the current cursor and extends to the word boundary (<kbd>w</kbd>) or typed character (<kbd>f</kbd>/<kbd>t</kbd>). You most likely want to append with <kbd>a</kbd> which inserts at the end of the selection and extends the selection to include the inserted text.
